Essentials/src/com/earth2me/essentials/MetaItemStack.java

```diff
@@ -338,6 +342,17 @@ public void addFireworkMeta(final CommandSource sender, final boolean allowShort
         }
     }
 
+    private Color getColorFromRgbHexString(String color) {
+        Color c;
+        try {
+            c = Color.fromRGB(Integer.decode(color.substring(color.indexOf('[') + 4, color.indexOf(']') - 1)));
```

Comment:
I know I normally advocate not declaring and initializing a variable that you use only on the next line, but this statement is sufficiently complicated that it might be worth doing.